modules/engage-paella-player-7/src/js/EpisodeConversor.js

```diff
@@ -293,41 +293,73 @@ function processAttachments(episode, manifest, config) {
   }
 }
 
-function getCaptions(episode) {
-  const result = [];
-  let attachments = episode.mediapackage?.attachments?.attachment;
-  if (!(attachments instanceof Array)) {
-    attachments = attachments ? [attachments] : [];
-  }
+function readCaptions(potentialNewCaptions, captions) {
+  potentialNewCaptions.forEach((potentialCaption) => {
+    try {
+      let captions_regex = /^captions\/([^+]+)(\+(.+))?/g;
+      let captions_match = captions_regex.exec(potentialCaption.type);
 
+      if (captions_match) {
+        let captions_lang = captions_match[3];
 
-  attachments.forEach(att => {
-    const exp = /captions\/([a-z\d]+)(?:\+([a-z]+))?/.exec(att.type);
-    if (exp) {
-      const format = exp[1];
-      const lang = exp[2] || '';
-      result.push({
-        id: att.id,
-        lang: lang,
-        text: lang || 'Unknown language',
-        format: format,
-        url:att.url
-      });
+        // TODO: read the lang from the dfxp file
+        //if (captions_format == "dfxp") {}
+
+        if (!captions_lang && potentialCaption.tags && potentialCaption.tags.tag) {
+          if (!(potentialCaption.tags.tag instanceof Array)) {
+            potentialCaption.tags.tag = [potentialCaption.tags.tag];
+          }
+          potentialCaption.tags.tag.forEach((tag)=>{
+            if (tag.startsWith('lang:')){
+              let split = tag.split(':');
+              captions_lang = split[1];
```

Comment:
Arguably simpler and only removes the prefix instead of splitting on every `:` character:
```suggestion
              captions_lang = tag.substring('lang:'.length);
```

Reply:
Inclined to agree, will change.